### PR TITLE
UCS: Handle differences in the signal handling on Linux vs. FreeBSD.

### PR DESCRIPTION
**Pull request by kostikbel**
## What
Makes signal handling code compilable on FreeBSD.

## Why ?
To continue merging FreeBSD port.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by dmitrygx** [comment]
ok to test

File: src/ucs/debug/debug.c [hunk 1]
```diff
@@ -1019,6 +1019,13 @@ static void* ucs_debug_get_orig_func(const char *symbol, void *replacement)
     return func_ptr;
 }
 
+#ifndef HAVE_SIGHANDLER_T
```
Comment (yosefe) [inline src/ucs/debug/debug.c]: can we do 
```
#if !HAVE_SIGHANDLER_T
#if HAVE___SIGHANDLER_T
...
```

File: src/ucs/async/signal.c [hunk 2]
```diff
@@ -243,7 +243,9 @@ static ucs_status_t ucs_async_signal_install_handler()
         new_action.sa_sigaction = ucs_async_signal_handler;
         sigemptyset(&new_action.sa_mask);
         new_action.sa_flags    = SA_RESTART|SA_SIGINFO;
+#ifdef HAVE_SIGACTION_SA_RESTORER
```
Comment (yosefe) [inline src/ucs/async/signal.c]: `#if`

File: src/ucs/debug/debug.c [hunk 3]
```diff
@@ -110,7 +110,9 @@ const char *ucs_signal_names[] = {
     [SIGSYS + 1] = NULL
 };
 
+#ifdef HAVE_SIGACTION_SA_RESTORER
```
Comment (yosefe) [inline src/ucs/debug/debug.c]: `#if`

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11479/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 30-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/a9708171d05ecc0c5cb3e332835686520eb55f9b/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/620cf90457de9e85da2873955d20e3fb8bfaca25/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/9cfa114114e0cc483f982b97f33cbc9cca85fae6/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/626054ff7d6a215a8694a8e5bf9b7c586d6b58a9/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/ef43870e4db461f36227e1ecd29dde40cc343a3a/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/047746f8feae5390e3af104939122c6b0f0c44fb/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/5c56249af26e44dd77b38209424aaec20e7e17e3/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/02c08359757e4d8f8fcfe80bd0ad1050e72fa3c6/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/22286d94c69ab13496e691e4a176d7d3eccacce5/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/81933a7cf2d9183aa99129ead1033c6a959eb268/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/25eb283cd072e6ee057d845c6ca51d3cbeb68d2a/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/5378f64bb9d7f922d36dddf7dbf6c152bd6330e8/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/c03be14374782d83f6f4a38a6a94cc8e63cdf190/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/47b73883a3d2fec4419f0d7c41350993ce490ec0/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/3bb1a57c6d7589d7dcde6bc8252f5fcf79539fc4/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/8b7e592c149533993b9f303a0dca211024aeb5d7/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/679aef8bb5354143e3f90dff0b945f0f12ace81c/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/125d44e87517aec17669daa0fc1a0fadfb80ab7b/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/ed21c8e3a80a812893916543b683f5a94b9470bb/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/9500dddfcbad15f1c1c7adba67ed5f4ad0144850/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/c80bff1a417f13a48f26ba1840ded8ad4a10ce25/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/bd9bfccdd2b8b6e5533cf3b007680a6da2f6d796/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/cbe1c935f780dcc5f1f7be4b6825831b4d5e3153/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/22abc910ac46191ab4f13110cade007dfcd28642/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/fa944da02571810a72dc7eb22d0da798b695216d/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/1e3a7988f2a7fed397583ea5a1ab5b8c052a88f5/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/54aa8eb3f53f681619cadba54a6082e13a0d4c9a/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/fcd7406fb0fe902b5494ebba8d1c21a4e4d64684/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/158a34b89215bedde6cc84c4b3062b2d9b918164/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/00dfe0112a9ecae0f2cc9d638849cbd5b663de61/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/110b1496217309b5370c24e5f0329731f4fc964b/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/5537897e0fdda01094648213ddfd298cf552e107/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/82bc51e87cc7190dec5ce4f6a236f5c1/raw/3b7427476ca3c2050b9f3fd9a6bd0140752daf46/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/ucs/debug/debug.c (hunk 3 above)
Comment (kostikbel) [reply src/ucs/debug/debug.c]: Done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11494/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 33 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Jul-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/e83bd310a5c07f218fc4d2241c8da8ca43031982/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/60bd86b5afc49df0eac745244466f073683e33b9/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/7c6635538bba204e71626f6294155adeebddfd38/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/19e482a9f97cdb5b2b0c673e29b7a6a56dae8e89/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/2a392d0f29597152504f9dba43e75e04a37b665c/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/bc9edfbb938bf16a1295efa07ab507db7ad98a75/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/b213d1a133b9d2ff865f305f8c9b886720799ac5/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/090287cd04e4d35bff32d22ffbf3f6ddd9ffd7f5/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/da81e56cde0bd56b2c47c98624f620f5550e25dc/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/f561b4e5288c70c132ca2814acd4a1afa95b954c/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/cf97edfeb41a2c1645c5621e0e641e3e41091b3a/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/796adbc7dd6877ffed323cb10d98b21175257183/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/bc459eff801e0b122891722936b958ecf66fa361/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/55c6c2d4358da98548432cacbfe6865bd314d9b9/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/c3626eb4d74c5e6cabd3ac22c04d71e09c587d5c/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/b36196ddc31c6e2072b3b2d2cfec6b8aa9efc23e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/93c1e2e2969bae25d6b7258080cae85435048709/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/90a0c521931bb9208ce1092025e166414316d3d3/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/13fecd8124966c82d206977e8cb834d934abeaf6/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/bc329e4b6245420e2f716dae08f96389d67f15b3/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/75a7a09ad46ff9835aadba71ef65f58f6fd16a3d/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/d3a532752e20b9b411fc67f1373d847ccdef8076/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/ed55f59d8241e6382d554a27bdfd03b4cea68464/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/b2182ff43043b48c0b5bb3d0c49a86c55868122f/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/eb3983333c48e5c8dffa91586100d05de9bbeee4/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/009cd3a4d6345d168901d6069fea457f549190d0/hpc-test-node-upstream_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/4aaecb1a480cddec4e61a7e41589b469c38b3ae9/hpc-test-node-upstream_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/0a041eff380962085acf086a2b3dc07e418ab2df/hpc-test-node-upstream_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-upstream_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/4f55a94753a0d0cf9b48b7cc00441ed759ca0df4/hpc-test-node-upstream_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/01af69383c08fcd8f42d0099f43f65b1e034a7b6/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/40626d5d4ef6960b727b4e6d7246cf6ab37d9851/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/d7f86689ef9e144e24ae485650490ef7b59e85fb/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/dc32ebe4977a52fec6f4b8e7917e155a/raw/9904546418c766aee249c8fcd3933436f0b6f3a8/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**